client/rpc/basic_client.go

```diff
@@ -0,0 +1,200 @@
+package rpc
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+
+	cmn "github.com/tendermint/tendermint/libs/common"
+	"github.com/tendermint/tendermint/rpc/client"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	"github.com/tendermint/tendermint/rpc/lib/client"
+	"github.com/tendermint/tendermint/types"
+
+	ntypes "github.com/binance-chain/go-sdk/common/types"
+	"github.com/binance-chain/go-sdk/types/tx"
+)
+
+const defaultTimeout = 5 * time.Second
+
+type Client interface {
+	cmn.Service
+	client.ABCIClient
+	client.SignClient
+	client.HistoryClient
+	client.StatusClient
+	EventsClient
+	DexClient
+	OpsClient
+}
+
+type EventsClient interface {
+	Subscribe(query string, outCapacity ...int) (out chan ctypes.ResultEvent, err error)
+	Unsubscribe(query string) error
+	UnsubscribeAll() error
+}
+
+func NewRPCClient(nodeURI string) *HTTP {
+	return NewHTTP(nodeURI, "/websocket")
+}
+
+type HTTP struct {
+	remote string
```

Comment:
What is the purpose of `remove` field? I found that you didn't use it.

Reply:
indeed useless. delete it.